src/dex/infusion/infusion.ts

```diff
@@ -0,0 +1,695 @@
+        tokenOut: to.address,
+        reservesIn: pairState.reserves1,
+        reservesOut: pairState.reserves0,
+        fee,
+        direction: false,
+        exchange: pair.exchange,
+        decimalsIn: from.decimals,
+        decimalsOut: to.decimals,
+        stable,
+      };
+    }
+    return {
+      tokenIn: from.address,
+      tokenOut: to.address,
+      reservesIn: pairState.reserves0,
+      reservesOut: pairState.reserves1,
+      fee,
+      direction: true,
+      exchange: pair.exchange,
+      decimalsIn: from.decimals,
+      decimalsOut: to.decimals,
+      stable,
+    };
+  }
+
+  async getPoolIdentifiers(
+    _from: Token,
+    _to: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    if (side === SwapSide.BUY) return [];
+
+    const from = this.dexHelper.config.wrapETH(_from);
+    const to = this.dexHelper.config.wrapETH(_to);
+
+    if (from.address.toLowerCase() === to.address.toLowerCase()) {
+      return [];
+    }
+
+    const tokenAddress = [from.address.toLowerCase(), to.address.toLowerCase()]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+
+    const poolIdentifier = `${this.dexKey}_${tokenAddress}`;
+    const poolIdentifierUniswap = poolIdentifier + this.poolPostfix(false);
+    const poolIdentifierStable = poolIdentifier + this.poolPostfix(true);
+    return [poolIdentifierUniswap, poolIdentifierStable];
+  }
+
+  poolPostfix(stable: boolean) {
+    return stable ? 'S' : 'V';
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  getAdapterParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    toAmount: NumberAsString, // required for buy case
+    data: InfusionData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: data.router,
+      payload: '',
+      networkFee: '0',
+    };
+  }
+
+  getDexParam(
+    src: Address,
+    dest: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: InfusionData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
+
+    let routerMethod: any;
+    let routerArgs: InfusionParam;
+    const stable = isStablePair(this.network, src, dest);
+
+    const from = isETHAddress(src)
+      ? this.dexHelper.config.data.wrappedNativeTokenAddress
+      : src;
+    const to = isETHAddress(dest)
+      ? this.dexHelper.config.data.wrappedNativeTokenAddress
+      : dest;
+
+    routerMethod = isETHAddress(src)
+      ? InfusionRouterFunctions.sellExactEth
+      : InfusionRouterFunctions.swapExactIn;
+    routerMethod = isETHAddress(dest)
+      ? InfusionRouterFunctions.sellExactToken
+      : routerMethod;
+
+    routerArgs = isETHAddress(src)
+      ? [
+          destAmount,
+          [{ from, to, stable }],
+          recipient,
+          Math.floor(new Date().getTime()) + 120,
```

Comment:
`getTime()` returns timestamp in milliseconds, Probably lost the division by 1000
```suggestion
          Math.floor(new Date().getTime() / 1000) + 120,
```

Reply:
done